### PR TITLE
[WIP] add ability to filter by groups for audit profiles

### PR DESCRIPTION
**Pull request by EmilyM1**
This allows a user the new field to enter the group name they would like to see the Audit Profiles on.

### CODE REVIEW

File: config/v1/types_apiserver.go [hunk 1]
```diff
@@ -88,7 +99,15 @@ type Audit struct {
...
 	// +kubebuilder:default=Default
-	Profile AuditProfileType `json:"profile,omitempty"`
+	// customRules specify profiles per group. These profile take precedence over the
+	// top-level profile field if they apply. They are evaluation from top to bottom and
+	// the first one that matches, applies.
+	//
+	// +listType=map
+	// +listMapKey=group
+	// +optional
+	Profile     AuditProfileType `json:"profile,omitempty"`
```
Comment (sttts) [inline config/v1/types_apiserver.go:109]: the profile field must go up. The go doc here is for CustomRules

File: config/v1/types_apiserver.go [hunk 2]
```diff
@@ -75,6 +75,17 @@ const (
...
 
+type AuditGroupType string
+
+const (
+	Group AuditProfileType = ""
+)
+
+type CustomRule struct {
+	Profile AuditProfileType `json:"profile,omitempty"`
+	Group   AuditGroupType   `json:"group,omitempty"`
```
Comment (sttts) [inline config/v1/types_apiserver.go:86]: I think we can just use string here.

Comment (sttts) [reply config/v1/types_apiserver.go:86]: Both fields need kubebuilder validation markers.

File: config/v1/types_apiserver.go [hunk 3]
```diff
@@ -88,7 +99,15 @@ type Audit struct {
 	//
 	// If unset, the 'Default' profile is used as the default.
 	// +kubebuilder:default=Default
-	Profile AuditProfileType `json:"profile,omitempty"`
+	// customRules specify profiles per group. These profile take precedence over the
+	// top-level profile field if they apply. They are evaluation from top to bottom and
+	// the first one that matches, applies.
+	//
+	// +listType=map
+	// +listMapKey=group
```
Comment (sttts) [inline config/v1/types_apiserver.go:107]: something is wrong here. Either the syntax of these markers is different or our controller-tools version is old. @EmilienM can you check in kubebuilder docs how to specify list types?

Comment (EmilyM1) [reply config/v1/types_apiserver.go:107]: Oh wow, I did not know there was another RH with a Git handle this close to mine.

**Comment by EmilienM** [comment]
> This allows a user the new field to enter the group name they would like to see the Audit Profiles on.

hi 😊

File: config/v1/types_apiserver.go (hunk 3 above)
Comment (EmilienM) [reply config/v1/types_apiserver.go:107]: > Oh wow, I did not know there was another RH with a Git handle this close to mine. 

hi 😊

Comment (EmilyM1) [reply config/v1/types_apiserver.go:107]: I believe maps is still valid:
https://book.kubebuilder.io/reference/markers/crd-processing.html?highlight=listType#crd-processing

what are your thoughts?

File: config/v1/types_apiserver.go (hunk 1 above)
Comment (EmilyM1) [reply config/v1/types_apiserver.go:109]: Will fix up

File: config/v1/types_apiserver.go (hunk 3 above)
Comment (sttts) [reply config/v1/types_apiserver.go:102]: Probably our controller-gen is too old, see https://github.com/openshift/build-machinery-go/blob/master/make/targets/openshift/controller-gen.mk#L11. Sync with @damemi how to update it.

Comment (damemi) [reply config/v1/types_apiserver.go:102]: @sttts do we still need our fork of controller-tools? Last update was in 2019 and the last release you published in February https://github.com/openshift/kubernetes-sigs-controller-tools/releases/tag/v0.2.5

Comment (EmilyM1) [reply config/v1/types_apiserver.go:102]: Also it could be a type error.

File: config/v1/types_apiserver.go (hunk 1 above)
Comment (EmilyM1) [reply config/v1/types_apiserver.go:104]: I have a question on this, is there a case where we will need standalone `Profile`?
So the question is:
Do we need
A) Profile

    Profile
    Group

or
B) Profile
    Group

File: config/v1/types_apiserver.go (hunk 3 above)
Comment (sttts) [reply config/v1/types_apiserver.go:102]: @damemi probably we only need it to host the binaries.

File: config/v1/types_apiserver.go [hunk 4]
```diff
@@ -88,6 +94,19 @@ type Audit struct {
...
 	// +kubebuilder:default=Default
+	// customRules specify profiles per group. These profile take precedence over the
+	// top-level profile field if they apply. They are evaluation from top to bottom and
+	// the first one that matches, applies.
+	//
+	// +listType=map
+	// +listMapKey=group
+	// +optional
+	CustomRules []CustomRule     `json:"group,omitempty"`
+//	Profile     AuditProfileType `json:"profile,omitempty"`
```
Comment (sttts) [inline config/v1/types_apiserver.go:105]: the profile must stay.

File: config/v1/types_apiserver.go (hunk 1 above)
Comment (sttts) [reply config/v1/types_apiserver.go]: The profile is always there. It defaults to `Default` if not explicitly set by the user.

So in the traditional case you only have the top-level profile. But with the new API you can add custom profiles per group.

**Comment by sttts** [comment]
OpenAPI diff looks incomplete. Maybe run `make update`?

**Comment by EmilyM1** [comment]
`make update` has been consistently failing unfortunately. Fails on some import related problems.

**Pull request closed without merging**